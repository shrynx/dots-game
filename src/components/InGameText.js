// @flow
import styled, { keyframes } from 'styled-components';
import { RED, YELLOW, CYAN, GREEN } from '../styles/colors';

const inGameTextAnimation = keyframes`
  0% {
    color: ${YELLOW};
  }
  25% {
    color: ${GREEN};
    transform: translate(-2px, 1px);
  }
  40% {
    color: ${YELLOW};
    text-shadow: none;
    transform: translate(0, 0);
  }
  50% {
    color: ${GREEN};
    text-shadow: 5px 5px 2px ${CYAN}, -5px -5px 2px ${RED};
    transform: translate(0px, 5px);
  }
  70% {
    color: ${YELLOW};
    text-shadow: none;
    transform: translate(0, 0);
  }
  80% {
    color: ${YELLOW};
    transform: translate(-2px, 1px);
  }
  100% {
    color: ${YELLOW};
    text-shadow: none;
  }
`;

const InGameText = styled.div`
  width: 100%;
  text-align: center;
  word-wrap: break-word;
  font-size: 60px;
  animation-name: ${inGameTextAnimation};
  animation-duration: 1s;
  animation-iteration-count: infinite;
`;

export default InGameText;
