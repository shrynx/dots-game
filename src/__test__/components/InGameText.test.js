import React from 'react';
import { render } from 'react-testing-library';
import InGameText from '../../components/InGameText';

it('InGameText component should render correctly', () => {
  const { container } = render(<InGameText>Game Over</InGameText>);
  expect(container).toMatchSnapshot();
});
