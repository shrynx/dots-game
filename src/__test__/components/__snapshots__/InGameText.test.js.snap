// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InGameText component should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa iWwDeN"
  >
    Game Over
  </div>
</div>
`;
